### ISSUE TITLE
Discussion performance

### ISSUE THREAD

**Issue by simon1222** [flarum/core#2838]
## Bug Report

**Current Behavior**
When I open the discussion with 126k comments , then flarum retrieves all posts. This behavior is too expensive, because response time is about 6s.

**Expected Behavior**
Load only required posts.

**Screenshots**
![image](https://user-images.githubusercontent.com/67348152/117173471-18265b80-adcd-11eb-8613-2f951d7ff577.png)
![image](https://user-images.githubusercontent.com/67348152/117173584-3429fd00-adcd-11eb-9897-8ed9ce49f1cd.png)

**Environment**
- Flarum version: beta 16
- Webserver: LiteSpeed
- Hosting environment: shared
- PHP version: 7.4

**Additional Context**
I had to turn off follow tags extension and remove loading of  tagState (1.2m rows in DB), because response time was much bigger than 6s. But this is issue for flarum/tag.

**Comment by askvortsov1** [comment]
> then flarum retrieves all posts. This behavior is too expensive, because response time is about 6s.

Not quite. It loads in **the ids** of all posts: the posts themselves are serialized only when the user scrolls to the appropriate point in the discussion.

I suppose it might be possible to load in the discussion without loading the post IDs, but there are some associated challenges. Also, there's the question of whether 125k post discussions are good UX: often, this might be better solved via a chat, or splitting the discussion into topics. The discourse people have some [good thoughts on this](https://meta.discourse.org/t/the-megatopic-public-good-or-public-menace/85592).

> I had to turn off follow tags extension and remove loading of tagState (1.2m rows in DB), because response time was much bigger than 6s. But this is issue for flarum/tag.

On the bright side, this is likely mostly fixed in the next release: see #2177

**Comment by luceos** [comment]
@simon1222 can you let us know whether this issue is resolved? Please re-open if not.

**Comment by rafaucau** [comment]
This discussion which contains 126k is still loading very slowly.
We have split it off into a new one.

It will be worth improving the loading of large discussions in Flarum at some point.
